Source/ExponentialBackoffRetryPolicy.swift

```diff
@@ -0,0 +1,180 @@
+            !retryableHTTPMethods.contains(httpMethod)
+        {
+            return (false, 0)
+        }
+
+        // Inspect the HTTP status code and error to see if the request should be retried
+        guard isRetryable(request.response, with: error) else { return (false, 0) }
+
+        // Compute time delay for the exponential backoff (0.5s, 1.0s, 2.0s, 4.0s, ...)
+        let timeDelay = pow(Double(exponentialBackoffBase), Double(request.retryCount)) * exponentialBackoffScale
+
+        return (true, timeDelay)
+    }
+
+    // MARK: - Private - Retry Helpers
+
+    private func isRetryable(_ response: HTTPURLResponse?, with error: Error) -> Bool {
+        if let statusCode = response?.statusCode, retryableStatusCodes.contains(statusCode) {
+            // Check if the response status code is retriable
+            return true
+        } else if let urlError = error as? URLError {
+            return isRetryable(urlError)
+        }
+
+        return false
+    }
+
+    private func isRetryable(_ urlError: URLError) -> Bool {
+        switch urlError.code {
+        case
+            //.appTransportSecurityRequiresSecureConnection,
+            .backgroundSessionInUseByAnotherProcess,
+            //.backgroundSessionRequiresSharedContainer,
+            .backgroundSessionWasDisconnected,
+            .badServerResponse,
+            //.badURL,
+            .callIsActive,
+            //.cancelled,
+            //.cannotCloseFile,
+            .cannotConnectToHost,
+            //.cannotCreateFile,
+            //.cannotDecodeContentData,
+            //.cannotDecodeRawData,
+            .cannotFindHost,
+            .cannotLoadFromNetwork,
+            //.cannotMoveFile,
+            //.cannotOpenFile,
+            //.cannotParseResponse,
+            //.cannotRemoveFile,
+            //.cannotWriteToFile,
+            //.clientCertificateRejected,
+            //.clientCertificateRequired,
+            //.dataLengthExceedsMaximum,
+            .dataNotAllowed, // Refers to cellular data
+            .dnsLookupFailed,
+            .downloadDecodingFailedMidStream,
+            .downloadDecodingFailedToComplete,
+            //.fileDoesNotExist,
+            //.fileIsDirectory,
+            //.httpTooManyRedirects,
+            .internationalRoamingOff,
+            .networkConnectionLost,
+            //.noPermissionsToReadFile,
+            .notConnectedToInternet,
+            //.redirectToNonExistentLocation,
+            //.requestBodyStreamExhausted,
+            //.resourceUnavailable,
+            .secureConnectionFailed,
+            .serverCertificateHasBadDate,
+            //.serverCertificateHasUnknownRoot,
+            .serverCertificateNotYetValid,
+            //.serverCertificateUntrusted,
+            .timedOut:
+            //.unknown,
+            //.unsupportedURL,
+            //.userAuthenticationRequired,
+            //.userCancelledAuthentication,
+            //.zeroByteResource,
```

Comment:
We spent an excruciating amount of time going through each of these in the original implementation. Some of these are a bit difficult to tell whether they are retriable or not.